pkg/executor/core/executor.go

```diff
@@ -156,3 +160,27 @@ func (ex Executor) verifyReference(ctx context.Context, subjectRef common.Refere
 
 	return types.VerifyResult{IsSuccess: isSuccess, VerifierReports: verifyResults}
 }
+
+func (ex Executor) addNestedVerifierResult(ctx context.Context, referenceDesc ocispecs.ReferenceDescriptor, subjectRef common.Reference, verifyResult *vr.VerifierResult) {
+	verifyParameters := e.VerifyParameters{
+		Subject:        fmt.Sprintf("%s@%s", subjectRef.Path, referenceDesc.Digest),
+		ReferenceTypes: []string{"*"},
+	}
+
+	// error is never returned from ex.VerifySubject, if an err occurs, the result contains the error message
+	nestedVerifyResult, err := ex.VerifySubject(ctx, verifyParameters)
+
+	if err != nil {
+		logrus.Errorf("Error verifying subject %s@%s: %v", subjectRef.Path, referenceDesc.Digest, err)
```

Comment:
if there is an error, should we also return a report with IsSuccess false ? similar to https://github.com/deislabs/ratify/blob/main/pkg/verifier/notaryv2/notaryv2.go#L133

Reply:
Updated in https://github.com/deislabs/ratify/pull/535/commits/99f48e8bc6cb1a3def8ecf0d626aeab166906b09 

The reason I chose to just log the error was because `ex.VerifySubject()` handles setting `IsSuccess: false` by calling  `executor.PolicyEnforcer.ErrorToVerifyResult(ctx, verifyParameters.Subject, err)`. `VerifySubject` never actually returns an error. It will always return `(result, nil)`. In my opinion, `VerifySubject` should either continue to handle the error itself and never return an error, or actually return the error so it can be handled appropriately in other parts of the code. Not sure if this is something we care about or if it should be tracked for future work. Thanks for the feedback!

Reply:
thanks for the clarification @joshuaphelpsms. Just to confirm my understanding, error encountered by VerifySubject will be included in verifyResult.NestedResults 's message?